internal/ledgerutil/identifytxs.go

```diff
@@ -0,0 +1,479 @@
+			blkstorage.IndexableAttrBlockHash,
+			blkstorage.IndexableAttrTxID,
+			blkstorage.IndexableAttrBlockNumTranNum,
+		},
+	}
+	metricsProvider := &disabled.Provider{}
+	// Create new block store provider
+	blockStoreProvider, err := blkstorage.NewProvider(conf, indexConfig, metricsProvider)
+	if err != nil {
+		return nil, err
+	}
+
+	return blockStoreProvider, nil
+}
+
+// txEntry represents and encapsulates a relevant transaction identified on the block store
+type txEntry struct {
+	TxID      string `json:"txid"`
+	BlockNum  uint64 `json:"blockNum"`
+	TxNum     uint64 `json:"txNum"`
+	TxVStatus string `json:"txValidationStatus"`
+	KeyWrite  string `json:"keyWrite"`
+}
+
+// Builds the transaction sets for each record by identifying and aggregating block store transactions containing the relevant records
+func findAndWriteTxs(blockStore *blkstorage.BlockStore, inputKeyMapWrapper *compKeyMapWrapper) (uint64, uint64, error) {
+	// Preparation for traversing block store
+	blockchainInfo, err := blockStore.GetBlockchainInfo()
+	if err != nil {
+		return 0, 0, err
+	}
+	// Get available block range for iterator
+	minBlockHeight := uint64(1)
+	// Check for first available block if block store is from bootstrapped peer
+	snapshotInfo := blockchainInfo.GetBootstrappingSnapshotInfo()
+	if snapshotInfo != nil {
+		minBlockHeight = snapshotInfo.GetLastBlockInSnapshot() + uint64(1)
+	}
+	maxBlockHeight := blockchainInfo.GetHeight() - 1
+	maxTxHeight := uint64(math.MaxUint64)
+	// Check if compKeyMap's highest record is earlier than block store's max block height
+	if inputKeyMapWrapper.maxBlockNum <= maxBlockHeight {
+		maxBlockHeight = inputKeyMapWrapper.maxBlockNum
+		maxTxHeight = inputKeyMapWrapper.maxTxNum
+	}
+	blocksItr, err := blockStore.RetrieveBlocks(minBlockHeight)
+	if err != nil {
+		return 0, 0, err
+	}
+	defer blocksItr.Close()
+
+	inputKeyMap := inputKeyMapWrapper.compKeyMap
+	// Set block index to minimum allowed block height
+	blockIndex := minBlockHeight
+	txIndex := uint64(0)
+	// Iterate through all completed blocks of block store
+	for blockIndex <= maxBlockHeight {
+		// Unpack next block
+		nextBlock, err := blocksItr.Next()
+		if err != nil {
+			return 0, 0, err
+		}
+		block := nextBlock.(*common.Block)
+		blockData := block.GetData().GetData()
+
+		// Iterate through block transactions
+		for txIndexInt, envBytes := range blockData {
+			txIndex = uint64(txIndexInt)
+			// Extract next transaction
+			env, err := protoutil.GetEnvelopeFromBlock(envBytes)
+			if err != nil {
+				return 0, 0, err
+			}
+			pl, err := protoutil.UnmarshalPayload(env.GetPayload())
+			if err != nil {
+				return 0, 0, err
+			}
+			ch, err := protoutil.UnmarshalChannelHeader(pl.Header.ChannelHeader)
+			if err != nil {
+				return 0, 0, err
+			}
+			txID := ch.GetTxId()
+
+			// Extract write set from transaction then iterate through transaction write set
+			res, err := protoutil.GetActionFromEnvelope(envBytes)
```

Comment:
This line fails for any config block, it returns error:
`Ledger Identify Transactions Error: error unmarshalling ProposalResponsePayload: proto: can't skip unknown wire type 7`

Before doing this you need to make sure the transaction is an endorser transaction. See an example of this here:
https://github.com/hyperledger/fabric/blob/main/core/ledger/kvledger/txmgmt/validation/batch_preparer.go#L230

Reply:
Right.